libs/@local/hash-graphql-shared/src/graphql/types.ts

```diff
@@ -17,7 +18,18 @@ export type TextToken =
       link?: string;
     }
   | { tokenType: "hardBreak" }
-  | { tokenType: "mention"; mentionType: "user" | "page"; entityId: EntityId };
+  | {
+      tokenType: "mention";
+      mentionType:
+        | "user"
+        | "page"
+        | "entity"
```

Comment:
@CiaranMn I don't see the need for the `"user"` and `"page"` mention types, as they require no additional data to render (just the `entityId`). The reason I didn't remove them is because this might break the app in unexpected ways for anyone that has inserted users and pages, as the mention type will be outdated.

Should I create a pre-launch follow-up task for removing these, so we can do it briefly before we wipe the database?

Reply:
I wouldn't worry about breaking existing mentions... potential reasons to keep distinct types:
1. if the user/page was somehow missing, we'd not know what type of mention was broken
2. we couldn't handle loading state differently, should we want to

Maybe create a task and we can triage it?

Reply:
I've created [the triage ticket](https://linear.app/hash/issue/H-902/consider-deprecating-user-and-page-mention-types-as-were-now-using), in this PR I've stopped making use of the `user` and `page` mentions and started using `entity` mentions directly instead - let me know if you think I should change this.

I agree we could handle the loading state differently, but there might be other ways of achieving that with the `entity` mention (by setting `loadingIcon` in the mention's metadata for example)